### PR TITLE
Beginning to change the functionality for the record and livestream buttons

### PR DESCRIPTION
**Pull request by joel-labbe**
Laying some ground work for the live stream and record buttons. If you want to point out stuff for me that would be great, but I know I am missing a good amount here.

### CODE REVIEW

File: src/raspberry_pi_driver/behaviors.py [hunk 1]
```diff
@@ -27,25 +31,57 @@ def intercom(pub, flag) -> None:
...
+        # Sleep thread for 5 seconds
+        sleep(5)
+        # Once recording is over and static url to link is given
+        pub.publish(json.dumps(["yt_livestream_link", yt_livestream_link]))
+    elif recording_on:
+        pass
+        # display message saying a video is being recorded
         #########################
         #   Missing code        #
         #########################
```
Comment (joel-labbe) [inline src/raspberry_pi_driver/behaviors.py]: A message here is redundant as it is already caught in livestream.py.

File: src/raspberry_pi_ui/buttons/livestream.py [hunk 2]
```diff
@@ -33,9 +36,80 @@ def __init__(self, button, pub, msg_q):
...
+                #########################
+                #   Missing code        #
+                #########################
+            else:
+                self.logger.info("Livestream won't turn off...")
+                set_button_property(self, "red", "Livestreaming...")
+                # display message saying to try again later
+                #########################
+                #   Missing code        #
+                #########################
```
Comment (joel-labbe) [inline src/raspberry_pi_ui/buttons/livestream.py]: This is really lengthy, I am trying to reduce this considerably at the moment.

**Comment by joel-labbe** [comment]
I think it is good enough to merge now. I tested it and it records a video, but fails on the second attempt because it thinks a camera resource is still being used. The live stream fails after trying to set it up because I didn't publish on youtube during the test, but the record light turns on so it should be good. I need to put in a flag that stops messages during live stream setup as it throws off the colors and causes issues.

**Comment by joel-labbe** [comment]
Forgot to call upload function after recording...

File: src/raspberry_pi_driver/behaviors.py [hunk 3]
```diff
@@ -2,8 +2,14 @@
 from time import sleep
 
 from src.raspberry_pi_alarm.buzzer_interface import Buzzer
+from src.raspberry_pi_camera.camera_interface import CameraInterface
 from src.raspberry_pi_motion_sensor.motion_interface import MotionPir
 
+livestream_on = False
+recording_on = False
+# Create camera object
+cam = CameraInterface()
```
Comment (FanchenBao) [inline src/raspberry_pi_driver/behaviors.py]: Suggestion: consider put `livestream_on` and `recording_on` in a NamedTuple in main function of `rpi_out_driver.py`. You can call it `camera_flags`.

Consider also instantiate `cam` in the same main function. Then, you can pass `camera_flags` and `cam` as arguments to your `record` and `livestream` behaviors.

This way, you can still modify the states of your flags, use `cam`, but we avoid global variables and make `behaviors.py` cleaner.

Comment (joel-labbe) [reply src/raspberry_pi_driver/behaviors.py]: Done.

**Pull request merged**